sw/host/opentitanlib/src/io/spi.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+
+/// Represents the SPI transfer mode.
+pub enum TransferMode {
+    Mode0,
+    Mode1,
+    Mode2,
+    Mode3,
+}
+
+/// Represents a SPI transfer.
+pub enum Transfer<'rd, 'wr> {
+    Read(&'rd mut [u8]),
+    Write(&'wr [u8]),
+    Both(&'rd mut [u8], &'wr [u8]),
+}
+
+/// A trait which represents a SPI Target.
+pub trait Target {
+    /// Gets the current SPI transfer mode.
+    fn get_transfer_mode(&self) -> Result<TransferMode>;
+    /// Sets the current SPI transfer mode.
+    fn set_transfer_mode(&mut self, mode: TransferMode) -> Result<()>;
+
+    /// Gets the current number of bits per word.
+    fn get_bits_per_word(&self) -> Result<u32>;
+    /// Sets the current number of bits per word.
+    fn set_bits_per_word(&mut self, bits_per_word: u32) -> Result<()>;
+
+    /// Gets the maximum allowed speed of the SPI bus.
+    fn get_max_speed(&self) -> Result<u32>;
+    /// Sets the maximum allowed speed of the SPI bus.
+    fn set_max_speed(&mut self, max_speed: u32) -> Result<()>;
+
+    /// Returns the maximum number of transfers allowed in a single transaction.
```

Comment:
Is this equivalent to the size of each transfer?

Reply:
It's the maximum size of the `Transfer` slice that this particular SPI host can process in a single SPI transaction (ie: period where chip select is held low).

Normally, reading a SPI eeprom would be something like:
`[Transfer::Write(&b'<the read opocde>'), Transfer::Read(&mut receive_buffer[0..256])]`

But you could perform a larger read in a single transaction by adding additional `Transfer::Read`s onto the slice.  The size of an individual transfer is also SPI controller specific... I think I'll need to send a follow-up PR to add an API for that.